crates/biome_js_analyze/src/lint/nursery/use_at_index.rs

```diff
@@ -0,0 +1,927 @@
+    member: AnyJsLiteralExpression,
+    object: AnyJsExpression,
+    option: &UseAtIndexOptions,
+) -> Option<UseAtIndexState> {
+    let AnyJsLiteralExpression::JsNumberLiteralExpression(member) = member else {
+        return None;
+    };
+    let value_token = member.value_token().ok()?;
+    let number = value_token.text_trimmed().parse::<i64>().ok()?;
+    (number >= 0 && option.check_all_index_access).then_some(UseAtIndexState::new(
+        make_number_literal(number),
+        ErrorType::Index { is_negative: false },
+        object,
+    ))
+}
+
+fn check_unary_expression_member(
+    member: JsUnaryExpression,
+    object: AnyJsExpression,
+    option: &UseAtIndexOptions,
+) -> Option<UseAtIndexState> {
+    if !option.check_all_index_access {
+        return None;
+    }
+    // ignore -5
+    let token = member.operator_token().ok()?;
+    if token.kind() == T![-] {
+        if let Some(arg) =
+            get_integer_from_literal(&solve_parenthesized_expression(member.argument().ok()?)?)
+        {
+            if arg >= 0 {
+                return None;
+            }
+        }
+    }
+    Some(UseAtIndexState::new(
+        AnyJsExpression::JsUnaryExpression(member),
+        ErrorType::Index { is_negative: false },
+        object,
+    ))
+}
+
+/// check hoge[0]
+fn check_computed_member_expression(
+    exp: &JsComputedMemberExpression,
+    option: &UseAtIndexOptions,
+) -> Option<UseAtIndexState> {
+    // check slice
+    if let Some(slice_err) =
+        analyze_slice_element_access(&AnyJsExpression::JsComputedMemberExpression(exp.clone()))
+    {
+        return Some(slice_err);
+    }
+    // invalid optional chain, mutable case
+    if exp.is_optional_chain()
+        || is_within_delete_expression(&AnyJsExpression::JsComputedMemberExpression(exp.clone()))
+            .unwrap_or(false)
+    {
+        return None;
+    }
+    // check member
+    let member = solve_parenthesized_expression(exp.member().ok()?)?;
+    let object = exp.object().ok()?;
+    match member.clone() {
+        // hoge[hoge.length - 1]
+        AnyJsExpression::JsBinaryExpression(binary) => {
+            check_binary_expression_member(binary, object, option)
+        }
+        // hoge[1]
+        AnyJsExpression::AnyJsLiteralExpression(literal) => {
+            check_literal_expression_member(literal, object, option)
+        }
+        // hoge[-x]
+        AnyJsExpression::JsUnaryExpression(unary) => {
+            check_unary_expression_member(unary, object, option)
+        }
+        AnyJsExpression::JsIdentifierExpression(_) => None,
+        _ if option.check_all_index_access => Some(UseAtIndexState::new(
+            member,
+            ErrorType::Index { is_negative: false },
+            object,
+        )),
+        _ => None,
+    }
+}
+
+fn check_call_expression_last(
+    call_exp: &JsCallExpression,
+    member: &JsStaticMemberExpression,
+) -> Option<UseAtIndexState> {
+    let args: Vec<_> = call_exp
```

Comment:
We could use `get_arguments_by_index`. See my other comment about `get_arguments_by_index`.